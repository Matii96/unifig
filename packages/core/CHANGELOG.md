# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2022-10-12)

### Bug Fixes

- **core:** added missing class-validator dependency ([6052855](https://github.com/Matii96/unifig/commit/60528552db7396b967907cd262cecb3612f7293a))

### Features

- **core:** improved manager interface ([cef08c7](https://github.com/Matii96/unifig/commit/cef08c77b380c5c063969821829adc309705f927))
- **core:** package setup ([cb2a066](https://github.com/Matii96/unifig/commit/cb2a066f4d40dfa31779cb187067f7134204ac3b))
- **core:** throwing exceptions upon encountering problems ([86b48d7](https://github.com/Matii96/unifig/commit/86b48d7166ba79a75755ade0a791271277d7c521))
- **env:** package setup ([5221aef](https://github.com/Matii96/unifig/commit/5221aeff7ac7b071619be5f6c9e537bb74cdacb8))

### BREAKING CHANGES

- **core:** changed getters names to more descriptive equivalents
