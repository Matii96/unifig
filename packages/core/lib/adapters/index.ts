export * from './config-adapter.interface';
export * from './plain.adapter';
