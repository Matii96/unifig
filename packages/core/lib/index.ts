export * from './adapters';
export * from './manager';
export * from './properties-mapping';
export * from './validator';
