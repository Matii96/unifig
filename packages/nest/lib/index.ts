export * from './injection';
export * from './config.module';
