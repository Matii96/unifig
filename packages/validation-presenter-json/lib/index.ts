export * from './to-json.options';
export * from './to-json';
