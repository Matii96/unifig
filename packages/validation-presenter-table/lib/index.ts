export * from './to-table.options';
export * from './to-table';
