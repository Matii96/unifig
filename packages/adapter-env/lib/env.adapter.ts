import { join } from 'path';
import { readFileSync } from 'fs';
import { sync } from 'fast-glob';
import { parse } from 'dotenv';
import { expand } from 'dotenv-expand';
import { ConfigSource, ConfigSyncAdapter } from '@unifig/core';
import { EnvConfigAdapterOptions } from './env.adapter.options';

/**
 * Loads configuration from object.
 */
export class EnvConfigAdapter implements ConfigSyncAdapter {
  private readonly _options!: EnvConfigAdapterOptions;

  constructor(options: EnvConfigAdapterOptions = {}) {
    this._options = options;
  }

  load(): ConfigSource {
    const config: ReturnType<typeof parse> = {};
    const paths = [...(this._options.envFilesPaths ?? []), join(process.cwd(), '.env')].flatMap(
      (path) => sync(path, { dot: true, unique: true }),
    );

    for (const envFilePath of paths) {
      this.parseEnvFile(config, envFilePath);
    }

    if (!this._options.ignoreEnvVars) {
      Object.assign(config, process.env);
    }

    return config;
  }

  private parseEnvFile(config: ReturnType<typeof parse>, envFilePath: string) {
    Object.assign(config, parse(readFileSync(envFilePath)));

    if (!this._options.expandVariables) {
      return;
    }

    const { parsed } = expand({ parsed: config, processEnv: {} });
    if (parsed) {
      Object.assign(config, parsed);
    }
  }
}
