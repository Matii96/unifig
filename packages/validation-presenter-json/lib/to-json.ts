import { ConfigValidationException } from '@unifig/core';
import { ToJSONOptions } from './to-json.options';

export const toJSON = (validationException: ConfigValidationException, options: ToJSONOptions = {}): string =>
  JSON.stringify(
    validationException.errors.map((templateError) => ({
      template: templateError.template.name,
      errors: templateError.errors,
    })),
    null,
    options.space
  );
