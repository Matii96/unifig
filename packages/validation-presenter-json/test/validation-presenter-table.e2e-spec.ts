import { Config, PlainConfigAdapter } from '@unifig/core';
import { toJSON } from '../lib';
import { TemplateMock } from './mocks';

describe('@unifig/validation-presenter-json (e2e)', () => {
  it('should format table from validation errors', async () => {
    const validationError = await Config.register({
      templates: [TemplateMock],
      enableImplicitConversion: false,
      adapter: new PlainConfigAdapter({ PORT: 'not-a-port', DB_PASSWORD: 'my-password' }),
    });
    expect(toJSON(validationError!)).toMatchSnapshot();
  });
});
