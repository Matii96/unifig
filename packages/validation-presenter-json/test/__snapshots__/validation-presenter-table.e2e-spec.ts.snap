// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@unifig/validation-presenter-json (e2e) should pretty print 2d errors in json format 1`] = `
"[
  {
    "template": "StorageOptions",
    "errors": [
      {
        "property": "port",
        "source": "PORT",
        "currentValue": "not-a-port",
        "failedConstraints": [
          {
            "details": "port must be an integer number",
            "name": "isInt"
          }
        ]
      },
      {
        "property": "db",
        "children": [
          {
            "property": "url",
            "source": "DB_URL",
            "failedConstraints": [
              {
                "details": "url must be a string",
                "name": "isString"
              }
            ]
          },
          {
            "property": "password",
            "source": "DB_PASSWORD",
            "failedConstraints": [
              {
                "details": "password must be a string",
                "name": "isString"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "template": "NetworkOptions",
    "errors": [
      {
        "property": "ipRange",
        "failedConstraints": [
          {
            "details": "ipRange must be defined",
            "name": "isDefined"
          }
        ],
        "children": []
      }
    ]
  }
]"
`;

exports[`@unifig/validation-presenter-json (e2e) should print 2d errors in json format in one line 1`] = `"[{"template":"StorageOptions","errors":[{"property":"port","source":"PORT","currentValue":"not-a-port","failedConstraints":[{"details":"port must be an integer number","name":"isInt"}]}]}]"`;
